src/amo/components/UserProfileEdit/index.js

```diff
@@ -155,10 +155,7 @@ export class UserProfileEditBase extends React.Component<Props, State> {
     }
 
     if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
-      this.setState({
-        pictureData: null,
-        successMessage: i18n.gettext('Profile successfully updated'),
-      });
+      router.push(`/${lang}/${clientApp}/user/${newUsername}/`);
```

Comment:
I think we should have a `return` below no?

Reply:
good to have

Reply:
I can do that tonight :)